src/extension/background-script/migrations/__tests__/legacylnurlauth.test.ts

```diff
@@ -0,0 +1,114 @@
+import db from "~/extension/background-script/db";
+import state from "~/extension/background-script/state";
+import type { Allowance } from "~/types";
+
+import migrate from "../index";
+
+describe("With lnurlAuth enabled allowance", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+    state.getState().reset();
+    db.allowances.toCollection().delete();
+  });
+
+  const mockAllowances: Allowance[] = [
+    {
+      enabled: true,
+      host: "numbergoup.ngu",
+      id: 1,
+      imageURL: "https://numbergoup.ngu/favicon.ico",
+      lastPaymentAt: "0",
+      lnurlAuth: true,
+      name: "ngu",
+      remainingBudget: 500,
+      totalBudget: 500,
+      createdAt: "123456",
+      payments: [],
+      paymentsAmount: 0,
+      paymentsCount: 0,
+      percentage: "0",
+      usedBudget: 0,
+      tag: "",
+    },
+  ];
```

Comment:
Hard to understand which data was altered for the test.

Why not define one basic object at the top of file like:

```TS
const allowance: Allowance = {
  enabled: true,
  host: "numbergoup.ngu",
  id: 1,
  imageURL: "https://numbergoup.ngu/favicon.ico",
  lastPaymentAt: "0",
  lnurlAuth: true,
  name: "ngu",
  remainingBudget: 500,
  totalBudget: 500,
  createdAt: "123456",
  payments: [],
  paymentsAmount: 0,
  paymentsCount: 0,
  percentage: "0",
  usedBudget: 0,
  tag: "",
};
```

And then for each test just spread in the data you need for this specific test run:

```TS
  const mockAllowances = [
    {
      ...allowances,
      lnurlAuth: false,
    },
  ];
```

... etc

Reply:
+1 thought about that, too. but thought we need general fixtures for that? (same is used in other places?)
how do we best load fixtures?

Reply:
There are several ways.
Why not add a `fixtures` folder next to the `types` folder?
And then `import {allowanceFixture } from 'scr/fixtures/allowances'` for example

Reply:
Good point. In the beginning we didn't have a lot of tests but now the amount of "fixtures" is growing and we could define them once and reuse them. I can help with that.